File: app/src/main/java/com/focus/mob/NiveauDistractionActivity.java
package com.focus.mob;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class NiveauDistractionActivity extends AppCompatActivity {

    private View[] cards;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_niveau_distraction);

        ImageButton btnBack = findViewById(R.id.btn_back);
        TextView tvPasser = findViewById(R.id.tv_passer);
        Button btnContinuer = findViewById(R.id.btn_continuer);

        cards = new View[]{
                findViewById(R.id.card_zen),
                findViewById(R.id.card_moyen),
                findViewById(R.id.card_distrait)
        };

        // Default selection
        selectCard(findViewById(R.id.card_moyen));

        for (View card : cards) {
            card.setOnClickListener(v -> selectCard(v));
        }

        btnBack.setOnClickListener(v -> finish());
        tvPasser.setOnClickListener(v -> navigateToNext());
        btnContinuer.setOnClickListener(v -> navigateToNext());
    }

    private void selectCard(View selectedCard) {
        for (View card : cards) {
            card.setSelected(card == selectedCard);
        }
    }

    private void navigateToNext() {
        Intent intent = new Intent(NiveauDistractionActivity.this, AutorisationsActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/focus/mob/ui/onboarding/NiveauDistractionActivity.kt
package com.focus.mob.ui.onboarding
import com.focus.mob.R

import com.focus.mob.ui.auth.*
import com.focus.mob.ui.main.*
import com.focus.mob.ui.onboarding.*
import com.focus.mob.ui.session.*
import com.focus.mob.utils.*

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.focus.mob.databinding.ActivityNiveauDistractionBinding

class NiveauDistractionActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNiveauDistractionBinding
    private lateinit var cards: Array<View>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityNiveauDistractionBinding.inflate(layoutInflater)
        setContentView(binding.root)

        cards = arrayOf(binding.cardZen, binding.cardMoyen, binding.cardDistrait)

        // Default selection
        selectCard(binding.cardMoyen)

        cards.forEach { card -> card.setOnClickListener { selectCard(it) } }

        binding.btnBack.setOnClickListener { finish() }
        binding.tvPasser.setOnClickListener { navigateToNext() }
        binding.btnContinuer.setOnClickListener { navigateToNext() }
    }

    private fun selectCard(selectedCard: View) {
        cards.forEach { it.isSelected = (it == selectedCard) }
    }

    private fun navigateToNext() {
        startActivity(Intent(this, AutorisationsActivity::class.java))
        fadeTransition()
    }
}
